commit b5032c3d37aa614644c7afbad33bb8226a52e6da
Author: Florian Weimer <fweimer@redhat.com>
Date:   Mon Feb 28 11:50:41 2022 +0100

    io: Add fsync call in tst-stat
    
    io/tst-stat and io/tst-stat-lfs fail sporadically on the Fedora
    builders, and this change hopefully helps to avoid the issue.
    
    (cherry picked from commit ae132284092edc5885315b44cd17d5ea91177e49)

diff --git a/io/tst-stat.c b/io/tst-stat.c
index 82e965de6ad87f61..be20cf16d70d05cc 100644
--- a/io/tst-stat.c
+++ b/io/tst-stat.c
@@ -69,6 +69,10 @@ do_test (void)
   TEST_VERIFY_EXIT (fd >= 0);
   support_write_file_string (path, "abc");
 
+  /* This should help to prevent delayed allocation, which may result
+     in a spurious stx_blocks/st_blocks difference.  */
+  fsync (fd);
+
   bool check_ns = support_stat_nanoseconds (path);
   if (!check_ns)
     printf ("warning: timestamp with nanoseconds not supported\n");
